pkg/cache/cache.go

```diff
@@ -138,6 +138,11 @@ func (sc *schedulerCache) WaitForCacheSync(stopCh <-chan struct{}) bool {
 
 // Assumes that lock is already acquired.
 func (sc *schedulerCache) addPod(pod *v1.Pod) error {
+	// only add Pending/Running pod
+	if pod.Status.Phase != v1.PodPending && pod.Status.Phase != v1.PodRunning {
+		return nil
+	}
+
```

Comment:
please refer to kube-scheduler on how to retrieve pending/running pods; apiserver provides the functionality to retrieve pods by state.phase.